### ISSUE TITLE
Some user friendly changes

### ISSUE THREAD

**Issue by bluescreen25** [VDJartnet/VDJartnet#13]
A wish to developers :
- exclude preset.txt and config.txt
- preset names in lines instead of full script if preset pushed to line
- full script accept in line if first letter is ~. So you can test live without close/open plugin and after that you can edit the config.txt and preset.txt  to add new preset with name. Normal user do not scripting on the fly. After some days no one remember which function a line has....

**Comment by aDifferentJT** [comment]
Thank you for your suggestions, in version 1.0.0 we intend to allow config lines to be given names and to have those names copy across with the presets.
If you wish to help with this or any other aspects of the project then you are welcome to submit a pull request at which point we will be more than happy to discuss your changes.

Furthermore, please note that I am a busy person with many things that I could be doing that would benefit me more than working on a piece of software that has brought me no compensation and I am unlikely to use to any significant extent. Despite this I have invested a significant amount of time in this project and, though I offer no guarantee of this fact, nor fixed timelines, I intend to continue to do so until such a time as I deem it stable and complete.

As I'm sure you will have noticed 
```
VDJartnet is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.
```
As such I would appreciate it if, when you ask that we extend or add to VDJartnet, you would treat us with due courtesy and respect, keeping in mind that VDJartnet is provided free to you, to use as you will.

**Comment by bluescreen25** [comment]
Hello, ok . You all have my respect for your work . Sorry my English is a bad and sometimes rough but in my mind , no one develops things which are given to public without to have successful experience through user. At the moment I make a lot of work with interesting script macros for the user in VDJ and QLC forum. It should push the project . Also my real meaning of development in advanthing things without testing the last version. I am so sorry that I can only Delphi and not C++. I would help you a lot. I think VDJArtnet can do things which a expensive box like Soundswitch for Serato/VDJ does in Autoloop mode. Must only be packed user friendly in plugin so that user does the work of cool macros and can exchange. That in future also as 64bit . I think this brings you much respect...
And on the way. I developed a car PC solution years ago with all features of the actual 2  DIN radios with skins and more. But now all new cars have these solution inside. All the time I have a forum and much user are there with ideas and peoples with photoshop knowledge to help with the skins. This was a very funny time and pushed enormly my Delphi learning. I come from Germany. There if no pressure, no solution , no learning....

best regards ralf 

⁣Gesendet mit TypeApp ​

**Comment by aDifferentJT** [comment]
Thank you for this, I do apologise for my aggressiveness, recently I have been rewriting a lot of the code hence the regression in functionality and stability but I feel that that is now pretty much done, also the releases on the GitHub releases page are not intended to be stable releases but were produced by continuous integration when I was setting that up, I shall probably remove them soon.

**Comment by bluescreen25** [comment]
Hello Jonathan,

all is ok.  It is fine you to see in qlc+ forum,too.

Instead of making a pull request i will send you my preset.txt as mail , if this way is ok for you,too. 

Before, i will make some tests with newest scripts i made and include it.

In code i have only one request. See red lines, i made a change:

If script values > 255 then DMX value should limited to DMX 255 instead DMX 0. 

As user create some scripts and makes for example a fade in or fade out with false calculating, then there come wierd action on DMX if higher values sends 0. 

The user would expect that the slider is up, not down.

 

best regards, Ralf

 

 

void CVDJartnet::updateDMXvalues() {

    if (m_Enable == 1) {

        bool updated = false;

 

        for (int i = 0; i < 512; i++) {

            if (!config->channelCommands[i].empty()) {

                double resultDouble = -1;

                SendCommand("set $VDJartnetSend 256");

                SendCommand(config->channelCommands[i].c_str());

                GetInfo("get_var $VDJartnetSend", &resultDouble);

                int resultInt = (int)round(resultDouble);

                if (resultInt >= 0 && resultInt <= 255) {

                    updated = artnet.setChannel(i, (uint8_t)resultInt);

                } else if (resultInt >= 0 && resultInt > 255) {

                    updated = artnet.setChannel(i, 255);

                } else { updated = artnet.setChannel(i, 0); 

                           }

            } else {

                updated = artnet.setChannel(i, 0);

            }

        }

 

        if (updated || skippedPackets > skipPacketLimit) {

            artnet.sendArtnetPacket(config->host, config->port);

            skippedPackets = 0;

        } else {

            skippedPackets += 1;

        }

    }

}

 

 

Von: nixCodeX [mailto:notifications@github.com] 
Gesendet: Dienstag, 23. Oktober 2018 10:28
An: VDJartnet/VDJartnet
Cc: bluescreen25; Author
Betreff: Re: [VDJartnet/VDJartnet] Some user friendly changes (#13)

 

Thank you for this, I do apologise for my aggressiveness, recently I have been rewriting a lot of the code hence the regression in functionality and stability but I feel that that is now pretty much done, also the releases on the GitHub releases page are not intended to be stable releases but were produced by continuous integration when I was setting that up, I shall probably remove them soon.

**Comment by aDifferentJT** [comment]
The reason why values larger than 255 are treated as 0 is that 256 is placed into $VDJartnetSend before running the user script so if the user hasn't entered a valid script for that channel 256 is received back and so we understand receiving a 256 to mean that there isn't a valid script so we want to send 0, also you seem to be checking that resultInt >= 0 and resultInt > 255 which is redundant. This would be significantly easier if you could submit a pull request as I could see exactly what you changed.

**Comment by flumeware** [comment]
DMX values are now limited to 255, all values above this are set to 255. This change will be included in v1.0.

**Comment by aDifferentJT** [comment]
I'm now closing this as it's no longer clear what the issue is, if you have any outstanding issues please open them up as separate issues
